base/common/src/main/java/com/netscape/certsrv/client/PKIClient.java

```diff
@@ -121,13 +133,51 @@ public Object marshall(Object request) throws Exception {
        return request;
    }
 
+    /**
+     * Unmarshall response object using custom mapping if available.
+     */
+    public <T> T unmarshall(HttpEntity entity, Class<T> clazz) throws Exception {
+
+        if (entity == null) {
+            return null;
+        }
+
+        String response = EntityUtils.toString(entity);
+        if (clazz.isInstance(response)) {
+            return clazz.cast(response);
+        }
```

Comment:
Hmm... here the `response` is a `String` which is unlikely to have a subclass to downcast to, unless the `clazz` itself is a `String` too? It might make more sense if `response` were an `Object` so it can be downcasted to anything, but I'm not familiar enough with the HTTP API.

Reply:
The return is of type T so a return of the response will warning for a unchecked cast. The EntityUtils will not return the response content as an object in this case and we will do more conversion.

I am not modifying now but re-evaluate while implementing the other methods.